src/main/plugins/org.dita.pdf2/xsl/fo/troubleshooting-elements.xsl

```diff
@@ -0,0 +1,205 @@
+<?xml version='1.0'?>
+
+<!-- 20170503 SCH: Add support for troubleshooting elements. -->
+
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+    xmlns:fo="http://www.w3.org/1999/XSL/Format"
+    xmlns:dita2xslfo="http://dita-ot.sourceforge.net/ns/200910/dita2xslfo"
+    xmlns:opentopic="http://www.idiominc.com/opentopic"
+    xmlns:opentopic-index="http://www.idiominc.com/opentopic/index"
+    exclude-result-prefixes="opentopic opentopic-index dita2xslfo"
+    version="2.0">
+
+    <!-- Determines whether to generate titles for task sections. Values are YES and NO. -->
+    <xsl:param name="GENERATE-TASK-LABELS">
+        <xsl:choose>
+            <xsl:when test="$antArgsGenerateTaskLabels='YES'"><xsl:value-of select="$antArgsGenerateTaskLabels"/></xsl:when>
+            <xsl:otherwise>NO</xsl:otherwise>
+        </xsl:choose>
+    </xsl:param>
+  
+    <xsl:template match="*[contains(@class, ' troubleshooting/troubleshooting ')]" mode="processTopic"
+                name="processTroubleshooting">
+    <fo:block xsl:use-attribute-sets="task">
+      <xsl:apply-templates select="." mode="commonTopicProcessing"/>
+    </fo:block>
+  </xsl:template>
+  
+  <!-- Deprecated, retained for backwards compatibility -->
+  <xsl:template match="*" mode="processTroubleshooting">
+    <xsl:call-template name="processTroubleshooting"/>
+  </xsl:template>
+  
+    <xsl:template match="*[contains(@class, ' troubleshooting/troublebody ')]">
+        <fo:block xsl:use-attribute-sets="troublebody">
+            <xsl:call-template name="commonattributes"/>
+            <xsl:apply-templates/>
+        </fo:block>
+    </xsl:template>
+    
+    <xsl:template match="*[contains(@class, ' troubleshooting/troubleSolution ')]">
+        <fo:block xsl:use-attribute-sets="troubleSolution">
+            <xsl:call-template name="commonattributes"/>
+            <xsl:apply-templates/>
+        </fo:block>
+    </xsl:template>
+
+    <xsl:template match="*[contains(@class, ' troubleshooting/condition ')]">
+        <fo:block xsl:use-attribute-sets="condition">
+            <xsl:call-template name="commonattributes"/>
+            <xsl:apply-templates select="." mode="dita2xslfo:task-heading">

```

Comment:
The `dita2xslfo:retrieve-task-heading` mode was put in place specifically for tasks, as a way to handle the bad original design of task labels defined twice (once in PDF, once for common, with different IDs). It should not be used here - the single variable ID should be used to retrieve the variable. Same comment for other uses below.